### PR TITLE
Fix golint errors in pkg/apis/core

### PR DESCRIPTION
**Pull request by mahendrakariya**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Fix golint errors in `pkg/apis/core`

**Which issue(s) this PR fixes**:

Ref #68026

**Special notes for your reviewer**:
This is my first ever PR to Kubernetes and I am particular sceptical about commit https://github.com/mahendrakariya/kubernetes/commit/75e9bfc168077fcb9346e334b59d60a2c997735b. Please do let me know if there is a better way to do it.

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/apis/core/annotation_key_constants.go [hunk 1]
```diff
@@ -26,7 +26,7 @@ const (
 	// PodPresetOptOutAnnotationKey represents the annotation key for a pod to exempt itself from pod preset manipulation
 	PodPresetOptOutAnnotationKey string = "podpreset.admission.kubernetes.io/exclude"
 
-	// MirrorAnnotationKey represents the annotation key set by kubelets when creating mirror pods
+	// MirrorPodAnnotationKey represents the annotation key set by kubelets when creating mirror pods
```
Comment (oleg-balunenko) [inline pkg/apis/core/annotation_key_constants.go]: Full stop at the end of the comment.

File: pkg/apis/core/json.go [hunk 2]
```diff
@@ -24,5 +24,8 @@ import "encoding/json"
 var _ = json.Marshaler(&AvoidPods{})
 var _ = json.Unmarshaler(&AvoidPods{})
 
+// MarshalJSON panics to prevent marshalling of internal structs
```
Comment (oleg-balunenko) [inline pkg/apis/core/json.go]: Full stop.

File: pkg/apis/core/json.go [hunk 3]
```diff
@@ -24,5 +24,8 @@ import "encoding/json"
...
 func (AvoidPods) MarshalJSON() ([]byte, error) { panic("do not marshal internal struct") }
-func (*AvoidPods) UnmarshalJSON([]byte) error  { panic("do not unmarshal to internal struct") }
+
+// UnmarshalJSON panics to prevent unmarshalling of internal structs
```
Comment (oleg-balunenko) [inline pkg/apis/core/json.go]: Full stop.

File: pkg/apis/core/objectreference.go [hunk 4]
```diff
@@ -23,12 +23,15 @@ import (
 	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
+// SetGroupVersionKind sets the API version and kind of the object reference
```
Comment (oleg-balunenko) [inline pkg/apis/core/objectreference.go]: Full stop.

File: pkg/apis/core/objectreference.go [hunk 5]
```diff
@@ -23,12 +23,15 @@ import (
...
 func (obj *ObjectReference) SetGroupVersionKind(gvk schema.GroupVersionKind) {
 	obj.APIVersion, obj.Kind = gvk.ToAPIVersionAndKind()
 }
 
+// GroupVersionKind returns the API version and kind of the object reference
```
Comment (oleg-balunenko) [inline pkg/apis/core/objectreference.go]: Full stop.

File: pkg/apis/core/objectreference.go [hunk 6]
```diff
@@ -23,12 +23,15 @@ import (
...
 func (obj *ObjectReference) GroupVersionKind() schema.GroupVersionKind {
 	return schema.FromAPIVersionAndKind(obj.APIVersion, obj.Kind)
 }
 
+// GetObjectKind returns the kind of object reference
```
Comment (oleg-balunenko) [inline pkg/apis/core/objectreference.go]: Full stop.

File: pkg/apis/core/register.go [hunk 7]
```diff
@@ -39,8 +39,12 @@ func Resource(resource string) schema.GroupResource {
 }
 
 var (
+	// SchemeBuilder object to register various known types
```
Comment (oleg-balunenko) [inline pkg/apis/core/register.go]: Full stop.

File: pkg/apis/core/register.go [hunk 8]
```diff
@@ -39,8 +39,12 @@ func Resource(resource string) schema.GroupResource {
...
 	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
-	AddToScheme   = SchemeBuilder.AddToScheme
+
+	// AddToScheme represents a func that can be used to apply all the registered
+	// funcs in a scheme
```
Comment (oleg-balunenko) [inline pkg/apis/core/register.go]: Full stop.

File: pkg/apis/core/resource.go [hunk 9]
```diff
@@ -20,35 +20,37 @@ import (
...
+func (rl *ResourceList) Memory() *resource.Quantity {
+	if val, ok := (*rl)[ResourceMemory]; ok {
 		return &val
 	}
 	return &resource.Quantity{Format: resource.BinarySI}
 }
 
-func (self *ResourceList) Pods() *resource.Quantity {
-	if val, ok := (*self)[ResourcePods]; ok {
+// Pods returns the list of pods
```
Comment (oleg-balunenko) [inline pkg/apis/core/resource.go]: Full stop.

File: pkg/apis/core/resource.go [hunk 10]
```diff
@@ -20,35 +20,37 @@ import (
...
+func (rl *ResourceList) Pods() *resource.Quantity {
+	if val, ok := (*rl)[ResourcePods]; ok {
 		return &val
 	}
 	return &resource.Quantity{}
 }
 
-func (self *ResourceList) StorageEphemeral() *resource.Quantity {
-	if val, ok := (*self)[ResourceEphemeralStorage]; ok {
+// StorageEphemeral returns the list of ephemeral storage volumes, if any
```
Comment (oleg-balunenko) [inline pkg/apis/core/resource.go]: Full stop.

File: pkg/apis/core/types.go [hunk 11]
```diff
@@ -3653,18 +3696,14 @@ type NodeSpec struct {
 	// The DynamicKubeletConfig feature gate must be enabled for the Kubelet to use this field
 	// +optional
 	ConfigSource *NodeConfigSource
-
-	// Deprecated. Not all kubelets will set this field. Remove field after 1.13.
-	// see: https://issues.k8s.io/61966
-	// +optional
-	DoNotUse_ExternalID string
```
Comment (liggitt) [inline pkg/apis/core/types.go]: separate out this change into its own PR, this isn't just a linting fix and affects round-trippability. We may not be able to drop this in v1 Nodes

File: pkg/apis/core/types.go [hunk 12]
```diff
@@ -4664,10 +4718,14 @@ type Secret struct {
 	Type SecretType
 }
 
+// MaxSecretSize represents the max secret size.
+// 1 * 1024 * 1024 = 1 MB
```
Comment (liggitt) [inline pkg/apis/core/types.go]: we don't need the math in the godoc, can just say (1MB)

File: pkg/apis/core/v1/zz_generated.conversion.go [hunk 13]
```diff
@@ -4836,7 +4836,7 @@ func autoConvert_v1_NodeSpec_To_core_NodeSpec(in *v1.NodeSpec, out *core.NodeSpe
 	out.Unschedulable = in.Unschedulable
 	out.Taints = *(*[]core.Taint)(unsafe.Pointer(&in.Taints))
 	out.ConfigSource = (*core.NodeConfigSource)(unsafe.Pointer(in.ConfigSource))
-	out.DoNotUse_ExternalID = in.DoNotUse_ExternalID
+	// WARNING: in.DoNotUse_ExternalID requires manual conversion: does not exist in peer-type
```
Comment (liggitt) [inline pkg/apis/core/v1/zz_generated.conversion.go]: move this to the separate PR for discussion

File: pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go [hunk 14]
```diff
@@ -159,12 +158,14 @@ func autoConvert_v1alpha1_KubeSchedulerConfiguration_To_config_KubeSchedulerConf
 	if err := Convert_v1alpha1_KubeSchedulerLeaderElectionConfiguration_To_config_KubeSchedulerLeaderElectionConfiguration(&in.LeaderElection, &out.LeaderElection, s); err != nil {
 		return err
 	}
-	if err := componentbaseconfigv1alpha1.Convert_v1alpha1_ClientConnectionConfiguration_To_config_ClientConnectionConfiguration(&in.ClientConnection, &out.ClientConnection, s); err != nil {
+	// TODO: Inefficient conversion - can we improve it?
```
Comment (liggitt) [inline pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go]: this is unexpected, revert these changes?

File: pkg/apis/core/types.go (hunk 11 above)
Comment (mahendrakariya) [reply pkg/apis/core/types.go]: I have reverted this commit.

File: pkg/apis/core/types.go (hunk 12 above)
Comment (mahendrakariya) [reply pkg/apis/core/types.go]: This is done!

File: pkg/apis/core/v1/zz_generated.conversion.go (hunk 13 above)
Comment (mahendrakariya) [reply pkg/apis/core/v1/zz_generated.conversion.go]: I have reverted this commit.

File: pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go (hunk 14 above)
Comment (mahendrakariya) [reply pkg/scheduler/apis/config/v1alpha1/zz_generated.conversion.go]: I have reverted this commit.

File: pkg/apis/core/types.go (hunk 11 above)
Comment (mahendrakariya) [reply pkg/apis/core/types.go]: As discussed on Slack, I have renamed this var and removed `_`.
